app/src/services/compound_service.ts

```diff
@@ -0,0 +1,109 @@
+import { Contract, Wallet, ethers, utils } from 'ethers'
+import { BigNumber, formatUnits, parseUnits } from 'ethers/utils'
+
+import { Token } from '../util/types'
+
+import { cBATAbi, cDaiAbi, cETHAbi, cUSDCAbi, cUSDTAbi, cWBTCAbi } from './compound_abi'
+
+class CompoundService {
+  contract: Contract
+  signerAddress: Maybe<string>
+  provider: any
+  exchangeRate: number
+  constructor(address: string, symbol: string, provider: any, signerAddress: Maybe<string>) {
+    const cTokenABI = CompoundService.getABI(symbol)
+    if (signerAddress) {
+      const signer: Wallet = provider.getSigner()
+      this.contract = new ethers.Contract(address, cTokenABI, provider).connect(signer)
+    } else {
+      this.contract = new ethers.Contract(address, cTokenABI, provider)
+    }
+    this.exchangeRate = 0
+    this.signerAddress = signerAddress
+    this.provider = provider
+  }
+
+  init = async () => {
+    this.exchangeRate = await this.calculateExchangeRate()
+  }
+
+  calculateSupplyRateAPY = async (): Promise<number> => {
+    const supplyRate: number = await this.contract.supplyRatePerBlock()
+    const ethMantissa = 1e18
```

Comment:
Should the mantissa be different for different token decimal amounts?

Reply:
yeah I was a bit confused about this too since the docs have an example for ETH but the 1e18 mantissa seems to give the correct APY for all tokens regardless of the decimals so just changed the name of the constant to make it more generic but left the value as is.